src/Security/Security.php

```diff
@@ -651,7 +651,7 @@ public function login($request = null, $service = Authenticator::LOGIN)
     {
         if ($request) {
             $this->setRequest($request);
-        } elseif ($request) {
+        } elseif (!$request) {
```

Comment:
` elseif ($this->getRequest())`

Reply:
Would you agree it should be `elseif (!$request && $this->getRequest())`?

Reply:
Why do you need to check if $request is falsey twice?

Reply:
No, nevermind, `if ($request)` should be hit anyway, so yep, just a sec :)